content/docs/latest/reference/developer-guides/build-tips-and-tricks.md

```diff
@@ -0,0 +1,100 @@
+
+1. **Create RAMDISK**: For the fastest build times, set up a RAMDISK for storage.
+2. **Install Docker Engine**: Set the Docker root-data to the RAMDISK mount point if using RAMDISK.
+3. **Clone the Flatcar Scripts**:
+   ```bash
+   git clone https://github.com/flatcar/scripts
+   ```
+   Clone the repository onto the RAMDISK for best speedups.
+4. **Run the SDK Container**:
+   ```bash
+   ./run_sdk_container -t
+   ```
+   This command downloads the latest Docker SDK container image, which is around 10GB.
+5. **Build Packages**:
+   ```bash
+   ./build_packages
+   ```
+6. **Build the Image**:
+   ```bash
+   ./build_image
+   ```
+7. **Convert Image for VM Testing**:
+   ```bash
+   ./image_to_vm.sh <params>
+   ```
+   Use `qemu-kvm` to create a VM instance and test the image.
+
+#### Additional Tips and Tricks
+
+- **Binary Packages**: Use `-U` with `./run_sdk_container` if you want to use binary packages, though full builds generally rebuild everything.
+- **Tmpfs**: Using Tmpfs can provide additional speedups. However, in some cases, using NVMe might not differ significantly.
+- **Parallel Fetch**: Enable `parallel-fetch` in Portage to download files in the background during the build.
+- **Rust/Go Module Download Bottleneck**: A significant bottleneck is the sequential download of Rust crates and Go modules. Parallel downloads or HTTP connection reuse could help speed this up.
+  ```bash
+  ./build_packages --fetchonly
+  ```
+  This command fetches all required packages. In one test, it took 24 minutes to fetch all 433 Rust crates.
+
+**Discussion and Further Insights**: For ongoing discussions and further insights on optimizing Flatcar builds, refer to the [GitHub issue #1550](https://github.com/flatcar/Flatcar/issues/1550).
+
+#### Optimization Insights
+
+- **Instrumenting Build Timeline**: Using tools like `otel-cli` can provide visibility into build delays.
+- **Cyclic Dependencies**: With nightly builds (4100.0.0 or newer), cyclic dependencies have been resolved, speeding up `build_packages` by ~30%. https://github.com/flatcar/scripts/pull/2340
+- **Binary Package Considerations**: Binary packages for releases build faster due to CDN availability, while nightly builds rely on a single server (bincache).
+- **Tmpfs in SDK Container**: To avoid messing with your host system, you can mount a tmpfs inside the SDK container at `/build/amd64-usr`.
+- **Emerging Package Statistics**: Tools like `genlop` can provide detailed build statistics from Portage logs.
+
+#### Example Builds
+
+- **Equinix Metal**: A suitable machine can reduce build time to within 1 hour. Ideal setups include modern development environments like the m3 or a3 systems.
+- **Build Log Insights**: A recent test build on a 24-core i9-13900HX laptop took less than 45 minutes for a full package build. Load is distributed unevenly, with some bottlenecks in dependency installation. Reordering builds could help maximize resource usage.
```

Comment:
Hmm, this is rather a usage of the binary cached already built packages, and not a full or rather close-to full source build which contradicts the first 1.5h - 2h part, we need a clearer workflow on how to reach this time and what optimizations were used.